File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& arr) {
        vector<int>pos,neg,ans;
        int n=arr.size();
        for(int i=0; i<n; i++){
            if(arr[i]>0) pos.push_back(arr[i]);
            else neg.push_back(arr[i]);
        }
        int i=0;
        int j=0;
        while(i<pos.size() && j<neg.size()){
            ans.push_back(pos[i++]);
            ans.push_back(neg[j++]);
        }
        //if still elements are rem then push back
        if(i<pos.size()) ans.push_back(pos[i++]);
        if(j<neg.size()) ans.push_back(neg[j++]);
        return ans;

    }
};

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] arr) {
     //  vector<int>pos,neg,ans;
     ArrayList<Integer>pos=new ArrayList<>();
     ArrayList<Integer>neg=new ArrayList<>();
     ArrayList<Integer>ans=new ArrayList<>();


        int n=arr.length;
        for(int i=0; i<n; i++){
            if(arr[i]>0) pos.add(arr[i]);
            else neg.add(arr[i]);
        }
        int i=0;
        int j=0;
        while(i<pos.size() && j<neg.size()){
            ans.add(pos.get(i++));
            ans.add(neg.get(j++));
        }
        //if still elements are rem then push back
        if(i<pos.size()) ans.add(pos.get(i++));
        if(j<neg.size()) ans.add(neg.get(j++));
//convert arraylist into int[]
        int[] result = new int[ans.size()];

        for(int k = 0; k < ans.size(); k++) {
            result[k] = ans.get(k);
        }

        return result;
    }
}